src/Http/Http.Abstractions/src/Extensions/UseMiddlewareExtensions.cs

```diff
@@ -107,7 +109,9 @@ public static class UseMiddlewareExtensions
                 return (RequestDelegate)invokeMethod.CreateDelegate(typeof(RequestDelegate), instance);
             }
 
-            var factory = Compile<object>(invokeMethod, parameters);
+            var factory = RuntimeFeature.IsDynamicCodeSupported
```

Comment:
@JamesNK can you make a change to this to use https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimefeature.isdynamiccodecompiled?view=net-7.0 instead?